erpnext/stock/doctype/delivery_trip/delivery_trip.js

```diff
@@ -66,6 +66,53 @@ frappe.ui.form.on('Delivery Trip', {
 		}
 	},
 
+	validate: function (frm) {
+		if (frm.is_new() && frm.doc.delivery_stops) {
+			frappe.call({
+				method: "erpnext.stock.doctype.delivery_trip.delivery_trip.validate_unique_delivery_notes",
+				args: {
+					delivery_stops: frm.doc.delivery_stops
+				},
+				callback: function (r) {
+					if (r) {
+						let confirm_message = `Atleast one of the entered Delivery Notes already exist in
+												the following Delivery Trips: ${r.message.join(", ")}.
+												Do you still want to continue?`;
+
+						frappe.confirm(
+							__(confirm_message),
+							function () { return; }, // If "Yes" is selected
+							function () {
+								// If "No" is selected
+								frappe.db.delete_doc(frm.doc.doctype, frm.doc.name);
+								frappe.set_route("List", frm.doc.doctype)
+							}
```

Comment:
Wait I just realized something here...

So this runs when you click save right?
With frappe call you then fork calls...

```
->validate ->  save doc   ->  updated doc -> confirm code
           \->validate_unique_delivery_note /
```

So if the server is fast enough to return your doc with the new name assigned everything works correctly... If it doesn't you might end up calling `frappe.db.delete_doc(frm.doc.doctype, frm.doc.name)` when frm.doc.name isn't set?

Reply:
Hm, that's true.

I'm not sure what's the best way around this. The document will eventually save, and if we don't delete it, it'll stay as a Draft trip with duplicate info.